server/tserver/src/main/java/org/apache/accumulo/tserver/ThriftClientHandler.java

```diff
@@ -0,0 +1,1812 @@
+import org.apache.accumulo.server.util.Halt;
+import org.apache.accumulo.server.zookeeper.TransactionWatcher;
+import org.apache.accumulo.tserver.ConditionCheckerContext.ConditionChecker;
+import org.apache.accumulo.tserver.RowLocks.RowLock;
+import org.apache.accumulo.tserver.data.ServerConditionalMutation;
+import org.apache.accumulo.tserver.log.TabletServerLogger;
+import org.apache.accumulo.tserver.metrics.TabletServerScanMetrics;
+import org.apache.accumulo.tserver.metrics.TabletServerUpdateMetrics;
+import org.apache.accumulo.tserver.scan.LookupTask;
+import org.apache.accumulo.tserver.scan.NextBatchTask;
+import org.apache.accumulo.tserver.scan.ScanParameters;
+import org.apache.accumulo.tserver.session.ConditionalSession;
+import org.apache.accumulo.tserver.session.MultiScanSession;
+import org.apache.accumulo.tserver.session.SessionManager;
+import org.apache.accumulo.tserver.session.SingleScanSession;
+import org.apache.accumulo.tserver.session.SummarySession;
+import org.apache.accumulo.tserver.session.UpdateSession;
+import org.apache.accumulo.tserver.tablet.CommitSession;
+import org.apache.accumulo.tserver.tablet.CompactionInfo;
+import org.apache.accumulo.tserver.tablet.Compactor;
+import org.apache.accumulo.tserver.tablet.KVEntry;
+import org.apache.accumulo.tserver.tablet.PreparedMutations;
+import org.apache.accumulo.tserver.tablet.ScanBatch;
+import org.apache.accumulo.tserver.tablet.Tablet;
+import org.apache.accumulo.tserver.tablet.TabletClosedException;
+import org.apache.hadoop.fs.FSError;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.Text;
+import org.apache.htrace.Trace;
+import org.apache.htrace.TraceScope;
+import org.apache.thrift.TException;
+import org.apache.zookeeper.KeeperException.NoNodeException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.cache.Cache;
+import com.google.common.collect.Collections2;
+
+public class ThriftClientHandler extends ClientServiceHandler implements TabletClientService.Iface {
+
+  private static final Logger log = LoggerFactory.getLogger(ThriftClientHandler.class);
+  private static final long MAX_TIME_TO_WAIT_FOR_SCAN_RESULT_MILLIS = 1000;
+  private static final long RECENTLY_SPLIT_MILLIES = 60 * 1000;
+  private final TabletServer server;
+  private final WriteTracker writeTracker = new WriteTracker();
+  private final RowLocks rowLocks = new RowLocks();
+  private final TransactionWatcher watcher;
+  private final VolumeManager fs;
+  private final SecurityOperation security;
+  private final SessionManager sessionManager;
+  private final TabletServerUpdateMetrics updateMetrics;
+  private final TabletServerScanMetrics scanMetrics;
+  private final TabletServerLogger logger;
+
+  public ThriftClientHandler(TabletServer server) {
+    super(server.getContext(), server.getWatcher(), server.getFileSystem());
+    this.server = server;
+    this.watcher = server.getWatcher();
```

Comment:
As stated earlier, you can get the stuff from the context and the super class:

```suggestion
    super(server.getContext(), new TransactionWatcher(server.getContext()), server.getContext().getVolumeManager());
    this.server = server;
    this.watcher = transactionWatcher;
```